core/trino-main/src/main/java/io/trino/operator/join/SortedPositionLinks.java

```diff
@@ -79,8 +81,7 @@ public int link(int from, int to)
             // make sure that from value is the smaller one
             if (comparator.compare(from, to) > 0) {
                 // _from_ is larger so, just add to current chain _to_
-                List<Integer> links = positionLinks.computeIfAbsent(to, key -> new IntArrayList());
-                links.add(from);
+                positionLinks.computeIfAbsent(to, (ignored) -> new IntArrayList()).add(from);
```

Comment:
If we want to avoid get/put additional hash/probe, would `compute` work better?
```
positionLinks.compute(to, (key, value) -> (value == null ? new IntArrayList() : value).add(item));
```

Reply:
Logically no, it shouldn't logically make a difference compared to `computeIfAbsent` (which is preferred over separate `get(int)` / `put(int, T)` calls because those _would_ re-probe the table). In practice `Int2ObjectOpenHashMap#compute` takes a `BiFunction<? super Integer, ? super T, ? extends T>` so that extra key boxing is certainly less than ideal in this context.